\Week 1 663380359-8 นางสาวรพีพรรณ ศรีบุญเรือง

Lab 1.1: Environment Setup & First Script

In [ ]:
print("Hello, Python Course!")

Hello, Python Course!


Lab 1.2: Working with Variables and Basic Data Types
- Objective: Students will define variables of different data types, perform simple operations, and display results.


2. Declare and print variables:


In [ ]:
Age = 21
GPA = 3.10
Favorite_Color = "Blue"
Student = True

print("Age = ",Age ,type(Age))
print("GPA = ",GPA,type(GPA))
print("Favorite_Color = ",Favorite_Color,type(Favorite_Color))
print("Student = ",Student,type(Student))

Age =  21 <class 'int'>
GPA =  3.99 <class 'float'>
Favorite_Color =  Blue <class 'str'>
Student =  True <class 'bool'>


3. Perform simple arithmetic:


In [ ]:
num1 = 2
num2 = 9
sum = num1 + num2
print("Sum = ",sum)

Sum =  13


4. String concatenation and f-strings:


In [ ]:
First_name = "Rapeephan"
Last_name = "Sriboonraung"
Full_name = First_name + " " + Last_name
print(f"Hello,{Full_name}")

Hello,Sappy Kampy


5. User input and type conversion:


In [ ]:
num = input("What your favorite number : ")
print("type of A is : ", type(num))
num = int(num)
result = num + 5
print(num," + 5 = ", result)

What your favorite number : 62
type of A is :  <class 'str'>
62  + 5 =  67


Activity 1.1: Class Discussion - Uses of Scripting
Format: Open discussion, possibly in small groups first, then full class.
Prompt: "Based on the introduction, where do you think scripting with Python could be applied in real-world scenarios? Think about tasks you do daily or things you've seen in other courses."
Expected outcomes: Students will brainstorm and share ideas like automating file operations, simple data analysis, web development, game scripting, scientific computing, etc. Instructor will guide the discussion to highlight Python's versatility.


คำตอบ: Real-World Python Scripting Examples
Boring Task Automation: * Renaming hundreds of photo files at once based on their creation date.

Automatically organizing a messy "Downloads" folder into subfolders (Documents, Images, Installers) based on file extensions.

Data Wrangling: * Combining multiple weekly Excel sales reports into one master spreadsheet.

Extracting specific text or tables from a batch of PDF invoices.

Web Interaction: * Writing a script to check a website every day for a price drop on a specific item (Web Scraping).

Automating social media posts across different platforms.

Creative and specialized Tools:

Automating tasks or creating plugins inside 3D software like Blender or Maya.

Programming smart home routines or simple robotics using a Raspberry Pi.

Activity 1.2: "Python Puzzlers" - Identifying Data Types & Output
Format: Individual or pair work, followed by a class review.
Instructions: Provide a set of small Python code snippets (on a handout or projected). For each snippet, students must:
Predict the output.
Identify the data type of specific variables after certain operations.
Explain why they chose that output/data type.


In [ ]:
# Puzzler 1
x = 10
y = 3.5
z = x + y
print(z)
print(type(z))


13.5
<class 'float'>




```
# This is formatted as code
```



ทำไมถึงเป็น type <class 'float'>: ในโค้ดบรรทัด x = 10 ตัวแปร x เป็นตัวเลขจำนวนเต็ม (Integer หรือ int) ส่วน y = 3.5 เป็นตัวเลขทศนิยม (Float)
เมื่อคุณนำ x และ y มาบวกกัน (z = x + y) Python จะทำการแปลงผลลัพธ์ให้เป็นทศนิยม (Float) โดยอัตโนมัติ เพื่อป้องกันไม่ให้ข้อมูลส่วนที่เป็นทศนิยมสูญหายไปครับ ดังนั้น 10 + 3.5 จึงเท่ากับ 13.5 และมีประเภทข้อมูลเป็น float

In [ ]:
# Puzzler 2
name = "Alice"
age = "30" # Note: This is a string
message = name + " is " + age + " years old."
print(message)
print(type(age))

Alice is 30 years old.
<class 'str'>



* **ได้ผลลัพธ์:** `Alice is 30 years old.`
* **ทำไมถึงเป็น type `<class 'str'>`:**
หากสังเกตที่บรรทัด `age = "30"` จะเห็นว่าตัวเลข 30 ถูกเขียนอยู่ภายใต้เครื่องหมายคำพูด (Double quotes `""`)
ในภาษา Python **ข้อมูลใดๆ ก็ตามที่ถูกครอบด้วยเครื่องหมายคำพูด จะถือว่าเป็นข้อความ (String หรือ `str`) ทันที** แม้ว่าข้างในนั้นจะเป็นตัวเลขก็ตาม มันจะถูกมองเป็นแค่ตัวอักษร '3' กับ '0' เรียงต่อกัน ไม่สามารถนำไปบวกหรือคำนวณทางคณิตศาสตร์ได้ครับ ดังนั้นเมื่อเช็ค `type(age)` จึงได้ผลลัพธ์ออกมาเป็น `<class 'str'>`

In [ ]:
# Puzzler 3
is_sunny = True
is_raining = False
print(is_sunny and is_raining)
print(type(is_sunny))

False
<class 'bool'>


ได้ผลลัพธ์: False และ <class 'bool'>

ทำไมถึงได้ผลลัพธ์นี้: 1. is_sunny and is_raining ได้ False: เพราะตัวดำเนินการ and (และ) มีเงื่อนไขว่า "ผลลัพธ์จะเป็นจริง (True) ได้เพียงกรณีเดียวเท่านั้น คือข้อมูลทั้งสองฝั่งต้องเป็น True ทั้งคู่" แต่ในโค้ดนี้ is_sunny เป็น True และ is_raining เป็น False เมื่อเจอ and เข้าไป ผลลัพธ์รวมจึงออกมาเป็น False ครับ
2. ทำไมถึงเป็น type <class 'bool'>: ตัวแปรที่มีค่าเป็น True หรือ False (ขึ้นต้นด้วยตัวพิมพ์ใหญ่ใน Python) จัดอยู่ในประเภทข้อมูลที่เรียกว่า Boolean (ตรรกศาสตร์) ตัว Python จึงย่อประเภทนี้ให้เราเห็นเป็น <class 'bool'>

In [ ]:
# Grabs the input and immediately converts it to an integer
num_int = int(input("Enter a number: "))

result = num_int + 5
print(result)

Enter a number: 10
15


ได้ผลลัพธ์: 15

ทำไมรอบนี้ทำงานได้ไม่แปลงเป็นประเภทอื่น: เพราะเราใส่ฟังก์ชัน int(...) ครอบ input() เอาไว้ตั้งแต่แรก ทำให้ค่าข้อความ "10" ที่ผู้ใช้พิมพ์เข้ามา ถูกแปลงร่างกลายเป็นตัวเลขจำนวนเต็ม 10 ทันที ตัวแปร num_int จึงมีประเภทข้อมูลเป็น int เมื่อนำไปบวกกับตัวเลข 5 ตรงๆ ในบรรทัดถัดไป Python จึงคำนวณทางคณิตศาสตร์ได้ถูกต้องและพิมพ์ผลลัพธ์ 15 ออกมาโดยไม่มีข้อผิดพลาด